fix(whiteboard): highlight tool colors

Arthurk12 opened the pull request:
### What does this PR do?

- [feat(whiteboard): make default yellow color more vibrant](https://github.com/bigbluebutton/bigbluebutton/pull/24165/commits/9a20f9b0a8e204c2d7a23b595b9494fff3e2b3cf)
  Updates the default yellow color from #FFFF00 to a more vibrant shade, improving visual distinction from the orange color and enhancing overall visibility in drawings and highlights.

<img width="2551" height="1295" alt="Screenshot from 2025-11-07 19-23-03" src="https://github.com/user-attachments/assets/17138433-a6f7-404e-8218-cb1ba50d3103" />


### Closes Issue(s)
Closes #none

### More

The playback will still display the old yellow color, which will be adjusted in a separate PR:
<img width="2551" height="1295" alt="Screenshot from 2025-11-07 19-28-32" src="https://github.com/user-attachments/assets/ffc0e836-09d3-4194-b431-645502f5b2df" />

Arthurk12 committed:
feat(whiteboard): make default yellow color more vibrant

Updates the default yellow color from #FFFF00 to a more vibrant shade,
improving visual distinction from the orange color and enhancing overall
visibility in drawings and highlights.

gustavotrott commented:
hey, I gave it a try, and I have to say the auto-selecting yellow behavior feels quite annoying. when I switch back from the highlighter to another tool, it keeps drawing in yellow instead of restoring my previous color, which feels confusing and unintuitive.

I believe Steve intentionally designed it the current way. the existing logic (black automatically becomes yellow when highlight tool is selected) is simple to understand and easy to predict. any alternative behavior tends to break that clarity and creates frustration when the color suddenly changes to something unexpected.

I also discussed it internally, and the general consensus was that the default behavior in Tldraw already provides the most consistent and user-friendly experience.

lfzawacki commented:
Yeah, the original behavior in TLDraw is understandable, but the inconsistent display of the color, e.g a black highlighter draws as yellow. They've expressed they don't want to change this and so we've tried to at least have the right color be displayed by changing the color to yellow whenever the user selects the highlighter. As I see we could solve it a few different ways:

* Do nothing and keep the old behavior
* Change color to yellow whenever we switch to highlighter. The color can be changed normally afterwards (this is the solution in 66f7dbb
* Have the default color for the highlighter be the same as the one for pencil (black). This is the most consistent option, but might be equally confusing to people just wanting the highlighter to default to yellow which is a pretty common color for this usage.

AFAIK we can't change the position of the yellow color in the pallete for the highlighter, so this is not an option

I'd like some input on how to solve this as the status quo was also reported as confusing by customers and TBH it looks like a bug in the application because of the incosistent display of the color.

PS.: I'm assuming the third commit is incontroversial as it only adds a more useful shade of yellow to the pencil pallete instead of two very similar oranges.

kepstin commented:
This change needs to be accompanied by a bump to the reported bbb version for recordings (set in bigbluebutton.yml) so the recording processing scripts can tell which set of colours need to be used.

antobinary committed:
chore: bump record-and-playback version to 3.0.17

@Arthurk12 's bbb-playback change 'feat(whiteboard): make default yellow color more vibrant' warrants a bump in the bbb_version.
https://github.com/bigbluebutton/bbb-playback/pull/318#issuecomment-3576790864

Bump record-and-playback's version to '3.0.17' so we can identify in which recordings those new colors are present.

Arthurk12 committed:
chore: Bump bbb-playback to 5.4.2

https://github.com/bigbluebutton/bbb-playback/releases/tag/v5.4.2

prlanzarin commented on bbb-playback.placeholder.sh:
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-git clone --branch v5.4.0 --depth 1 https://github.com/bigbluebutton/bbb-playback bbb-playback
+git clone --branch v5.4.2 --depth 1 https://github.com/bigbluebutton/bbb-playback bbb-playback
```
<img width="640" height="292" alt="image" src="https://github.com/user-attachments/assets/769ac7a3-c2fe-4078-ac22-7b43d69dacfa" />
